transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java

```diff
@@ -416,37 +504,120 @@ public void onError(@NonNull Errors errors) {
+                    if (prepaidCardList.size() > 1) {
+                        sortPrepaidCard(prepaidCardList);
+                    }
+                    for (PrepaidCard prepaidCard : prepaidCardList) {
+                        TransferSourceWrapper sourceWrapper = new TransferSourceWrapper();
+                        sourceWrapper.setToken(prepaidCard.getField(TOKEN));
+                        sourceWrapper.setType(PREPAID_CARD);
+                        sourceWrapper.setIdentification(prepaidCard);
+                        sources.add(sourceWrapper);
+                    }
+                }
+                if (isWalletModel) {
+                    mSelectedTransferSource.postValue(sources.get(0));
+                } else if (isCardModel) {
+                    sources.remove(0);
+                    mSelectedTransferSource.postValue(sources.get(0));
+                    mSourceToken = sources.get(0).getToken();
+                    loadTransferDestination(mSourceToken);
+                }
+                mTransferSources.postValue(sources);
+            }
+
+            @Override
+            public void onError(@NonNull Errors errors) {
+                mIsLoading.postValue(Boolean.FALSE);
+                mLoadTransferRequiredDataErrors.postValue(new Event<>(errors));
+            }
+        });
+    }
 
     private boolean hasTransferAmountChanged(@Nullable final Transfer transfer) {
         return mTransferAvailableFunds.getValue() && transfer != null && !TextUtils.equals(
                 transfer.getDestinationAmount(), mTransferAmount.getValue());
     }
 
+    public ProgramModel getProgramModel() {
+
+        Hyperwallet.getDefault().getConfiguration(new HyperwalletListener<Configuration>() {
+            @Override
+            public void onSuccess(@Nullable Configuration result) {
+                result.getProgramModel();
+                System.out.println("Program Model " + result.getProgramModel());
+                mProgramModel = ProgramModel.valueOf(result.getProgramModel());
+            }
+
+            @Override
+            public void onFailure(HyperwalletException exception) {
+            }
+
+            @Override
+            public Handler getHandler() {
+                return null;
+            }
+        });
+        return mProgramModel;
+    }
+
+    private void sortPrepaidCard(List<PrepaidCard> prepaidCards) {
+        Collections.sort(prepaidCards, new Comparator<PrepaidCard>() {
+            @Override
+            public int compare(PrepaidCard firstPrepaid, PrepaidCard secondPrepaid) {
+                if (firstPrepaid.getPrimaryCardToken() != null || secondPrepaid.getPrimaryCardToken() == null) {
+                    return 1;
+                }
+                if (firstPrepaid.getPrimaryCardToken() == null || secondPrepaid.getPrimaryCardToken() != null) {
+                    return -1;
+                }
+                return 0;
+            }
+        });
+    }
```

Comment:
this should be one liner, below is in Kotlin but should be similar for java

`prepaidCards?.sortedWith(compareBy { it.primaryCardToken })`

Reply:
in java 8 we have Collections.sort(prepaidCards,
Comparator.nullsFirst(Comparator.comparingInt(PrepaidCard::getSize))); but it support from Android N.